### PR TITLE
Migrate kde, lmnn , rl and rewardclipping test to catch

### PR DESCRIPTION
**Pull request by jeffin143**
Refer #2523 

Scrapped out 6 more tests, now 36 left :) 

addding_test

### CODE REVIEW

File: src/mlpack/tests/kde_test.cpp [hunk 1]
```diff
@@ -921,17 +918,17 @@ BOOST_AUTO_TEST_CASE(NotTrained)
   KDE<>::Tree queryTree(query, oldFromNew);
 
   // Check results.
-  BOOST_REQUIRE_THROW(kde.Evaluate(query, estimations), std::runtime_error);
-  BOOST_REQUIRE_THROW(kde.Evaluate(&queryTree, oldFromNew, estimations),
+  REQUIRE_THROWS_AS(kde.Evaluate(query, estimations), std::runtime_error);
+  REQUIRE_THROWS_AS(kde.Evaluate(&queryTree, oldFromNew, estimations),
```
Comment (zoq) [inline src/mlpack/tests/kde_test.cpp:922]: I can't comment on the actual line, but can you fix the indentation in line 923.

File: src/mlpack/tests/lmnn_test.cpp [hunk 2]
```diff
@@ -145,13 +142,13 @@ BOOST_AUTO_TEST_CASE(LMNNInitialEvaluationTest)
   double objective = lmnnfn.Evaluate(arma::eye<arma::mat>(2, 2));
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(objective, 9.456, 1e-5);
+  REQUIRE(objective == Approx( 9.456).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:145]:
```suggestion
  REQUIRE(objective == Approx(9.456).epsilon(1e-7));
```

Remove the extra space here.

File: src/mlpack/tests/lmnn_test.cpp [hunk 3]
```diff
@@ -165,16 +162,16 @@ BOOST_AUTO_TEST_CASE(LMNNInitialGradientTest)
   lmnnfn.Gradient(coordinates, gradient);
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.288, 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(1, 0), 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(0, 1), 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 12.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.288).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:165]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.288).epsilon(1e-7));
```

Remove the extra space here.

File: src/mlpack/tests/lmnn_test.cpp [hunk 4]
```diff
@@ -165,16 +162,16 @@ BOOST_AUTO_TEST_CASE(LMNNInitialGradientTest)
...
+  REQUIRE(gradient(1, 0) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(0, 1) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(1, 1) == Approx( 12.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:168]:
```suggestion
  REQUIRE(gradient(1, 1) == Approx(12.0).epsilon(1e-7));
```

Remove the extra space here.

File: src/mlpack/tests/lmnn_test.cpp [hunk 5]
```diff
@@ -188,18 +185,18 @@ BOOST_AUTO_TEST_CASE(LMNNInitialEvaluateWithGradientTest)
   double objective = lmnnfn.EvaluateWithGradient(coordinates, gradient);
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(objective, 9.456, 1e-5);
+  REQUIRE(objective == Approx( 9.456).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:188]:
```suggestion
  REQUIRE(objective == Approx(9.456).epsilon(1e-7));
```

Remove the extra space here.

File: src/mlpack/tests/lmnn_test.cpp [hunk 6]
```diff
@@ -188,18 +185,18 @@ BOOST_AUTO_TEST_CASE(LMNNInitialEvaluateWithGradientTest)
...
   // Check Gradient
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.288, 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(1, 0), 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(0, 1), 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 12.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.288).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(0, 1) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(1, 1) == Approx( 12.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:193]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.288).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).margin(1e-5));
  REQUIRE(gradient(0, 1) == Approx(0.0).margin(1e-5));
  REQUIRE(gradient(1, 1) == Approx(12.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 7]
```diff
@@ -210,18 +207,18 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableObjectiveTest)
...
-  BOOST_REQUIRE_CLOSE(lmnnfn.Evaluate(coordinates, 2, 1), 1.576, 1e-5);
-  BOOST_REQUIRE_CLOSE(lmnnfn.Evaluate(coordinates, 3, 1), 1.576, 1e-5);
-  BOOST_REQUIRE_CLOSE(lmnnfn.Evaluate(coordinates, 4, 1), 1.576, 1e-5);
-  BOOST_REQUIRE_CLOSE(lmnnfn.Evaluate(coordinates, 5, 1), 1.576, 1e-5);
+  REQUIRE(lmnnfn.Evaluate(coordinates, 0, 1) == Approx( 1.576).epsilon(1e-7));
+  REQUIRE(lmnnfn.Evaluate(coordinates, 1, 1) == Approx( 1.576).epsilon(1e-7));
+  REQUIRE(lmnnfn.Evaluate(coordinates, 2, 1) == Approx( 1.576).epsilon(1e-7));
+  REQUIRE(lmnnfn.Evaluate(coordinates, 3, 1) == Approx( 1.576).epsilon(1e-7));
+  REQUIRE(lmnnfn.Evaluate(coordinates, 4, 1) == Approx( 1.576).epsilon(1e-7));
+  REQUIRE(lmnnfn.Evaluate(coordinates, 5, 1) == Approx( 1.576).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:215]:
```suggestion
  REQUIRE(lmnnfn.Evaluate(coordinates, 0, 1) == Approx( 1.576).epsilon(1e-7));
  REQUIRE(lmnnfn.Evaluate(coordinates, 1, 1) == Approx( 1.576).epsilon(1e-7));
  REQUIRE(lmnnfn.Evaluate(coordinates, 2, 1) == Approx(1.576).epsilon(1e-7));
  REQUIRE(lmnnfn.Evaluate(coordinates, 3, 1) == Approx(1.576).epsilon(1e-7));
  REQUIRE(lmnnfn.Evaluate(coordinates, 4, 1) == Approx(1.576).epsilon(1e-7));
  REQUIRE(lmnnfn.Evaluate(coordinates, 5, 1) == Approx(1.576).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 8]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 0, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:238]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 9]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 1, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:245]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 10]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 2, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:252]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 11]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 3, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:259]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 12]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 4, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:266]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

File: src/mlpack/tests/lmnn_test.cpp [hunk 13]
```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
...
   lmnnfn.Gradient(coordinates, 5, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:273]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 14]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
 
   double objective = lmnnfn.EvaluateWithGradient(coordinates, 0, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(objective, 1.576, 1e-5);
+  REQUIRE(objective == Approx( 1.576).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:293]:
```suggestion
  REQUIRE(objective == Approx(1.576).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 15]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:298]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 16]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
   objective = lmnnfn.EvaluateWithGradient(coordinates, 1, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(objective, 1.576, 1e-5);
+  REQUIRE(objective == Approx( 1.576).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:302]:
```suggestion
  REQUIRE(objective == Approx(1.576).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 17]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:307]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 18]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
   objective = lmnnfn.EvaluateWithGradient(coordinates, 2, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(objective, 1.576, 1e-5);
+  REQUIRE(objective == Approx( 1.576).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:311]:
```suggestion
  REQUIRE(objective == Approx(1.576).epsilon(1e-7));
```

File: src/mlpack/tests/lmnn_test.cpp [hunk 19]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:316]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 20]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
   objective = lmnnfn.EvaluateWithGradient(coordinates, 3, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(objective, 1.576, 1e-5);
+  REQUIRE(objective == Approx( 1.576).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:320]:
```suggestion
  REQUIRE(objective == Approx(1.576).epsilon(1e-7));
```

Remove the extra space.

File: src/mlpack/tests/lmnn_test.cpp [hunk 21]
```diff
@@ -293,61 +290,61 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableEvaluateWithGradientTest)
...
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```
Comment (zoq) [inline src/mlpack/tests/lmnn_test.cpp:325]:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

**Comment by zoq** [comment]
The failed test, seems unrelated, let me rerun the job.

**Review by birm — APPROVED** [review]

**Review by zoq — APPROVED** [review]
AlarmedReliableAcouchi

**Comment by zoq** [comment]
Thanks, looks like we a super close.

**Pull request merged**